packages/repository-json-schema/src/__tests__/unit/build-schema.unit.ts

```diff
@@ -232,13 +232,27 @@ describe('build-schema', () => {
       expect(key).to.equal('modelPartial');
     });
 
-    it('returns concatenated option names otherwise', () => {
+    it('returns "optional[id,_rev]" when a single option "optional" is set', () => {
+      const key = buildModelCacheKey({optional: ['id', '_rev']});
+      expect(key).to.equal('modelOptional[id,_rev]');
+    });
+
+    it('does not include "optional" in concatenated option names if it is empty', () => {
       const key = buildModelCacheKey({
-        // important: object keys are defined in reverse order
         partial: true,
+        optional: [],
         includeRelations: true,
       });
       expect(key).to.equal('modelPartialWithRelations');
     });
+
+    it('returns concatenated option names otherwise', () => {
+      const key = buildModelCacheKey({
+        partial: true,
+        optional: ['id', '_rev'],
```

Comment:
When `partial` is set to `true`, the `optional` config is effectively ignored, because all properties are optional. 

Should we tell the user that they are may be doing something wrong? Personally, I would throw an error if both `partial` and `optional` are set.

Regardless of the solution we choose, please add a test to verify what happens when both `partial` and `optional` are set.

---

On Slack, we discussed an alternative solution: instead of adding `optional?: (keyof T)[];`, we can modify `partial` to accept `boolean | keyof T)[]`. This removes the edge case where both optional and partial are set. My concern was that this solution may be more difficult to understand for our users.

Reply:
The reason I opted to use `optional` is because I felt the name `partial` might be confusing. If you set `partial` to `[prop1, prop2]` then it might make it seem like these are the only two properties required and the rest are optional. What do you think?

Reply:
I agree with @nabdelgadir that `partial: ['a', 'b']` can be misleading.

IMO, `partial: true` becomes a shortcut of `optional` to mark all properties optional. My intuitive interpretation is the more specific one wins, i.e., `optional: ['a', 'b']` should override `partial: true`. WDYT?

Reply:
I agree. I think `optional` should override `partial` as well.

Reply:
> I agree. I think `optional` should override `partial` as well.

Makes sense 👍 

I feel it's important to describe this rule in (API) docs and have a test to verify the behavior.